app/models/mixins/service_ansible_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ServiceAnsibleMixin
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def with_applied_config_info(actions, config_info, opts = nil)
```

Comment:
I am not a fan of using a constant if it is used only once for better readability. In this case I would do something like
```
def with_applied_config_info(config_info, opts = nil)
  [:provision, :retirement, :reconfigure].each do |action|
  ...
end
```